frontend/apps/pwa/src/features/former/FieldSections.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+
+import { FormField } from '../../reducers/Former/types';
+
+import { FieldTypePicker } from './FieldTypePicker';
+import { FormerField } from './FormerField';
+import { FormerProps } from './types';
+
+type FieldSectionsProps = Omit<
+  FormerProps,
+  'fieldOptions' | 'formName' | 'ownerFormName' | 'saveForm' | 'setFormName'
+>;
+
+export const FieldSections: React.FC<FieldSectionsProps> = (props) => {
+  const [isAddingField, setIsAddingField] = useState(false);
+
+  const {
+    addField,
+    addOption,
+    cancelField,
+    fields,
+    openSubForm,
+    removeOption,
+    saveField,
+    selectedFieldId,
+    setFieldDescription,
+    setFieldIsKey,
+    setFieldName,
+    setFieldOption,
+    setFieldReferencedDatabaseId,
+    setFieldReferencedFormId,
+    setFieldRequired,
+    setSelectedField,
+  } = props;
+
+  if (isAddingField) {
+    return (
+      <FieldTypePicker
+        onCancel={() => {
+          setIsAddingField(false);
+        }}
+        onSubmit={(fieldKind) => {
+          addField(fieldKind);
+          setIsAddingField(false);
+        }}
+      />
+    );
+  }
+  return (
+    <div>
+      <div>
+        <button
+          className="btn btn-primary my-2 mb-3 w-100"
+          onClick={() => setIsAddingField(true)}
+        >
+          Add field
+        </button>
+      </div>
+      {fields.map((f: FormField) => (
+        <FormerField
+          key={f.id}
+          addOption={() => addOption(f.id)}
+          cancel={() => cancelField(f.id)}
+          errors={f.errors}
+          fieldDescription={f.description}
+          fieldIsKey={f.key}
+          fieldName={f.name}
+          fieldOptions={f.options}
+          fieldRequired={f.required}
+          fieldType={f.fieldType}
+          isSelected={f.id === selectedFieldId}
+          openSubForm={() => openSubForm(f.id)}
```

Comment:
I'd typically define a function:
```
const handleOpenSubForm = (id: string) => () => openSubForm(id);
```
Then:
```
openSubForm={handleOpenSubForm(f.id)}
```

Reply:
As I scrolled down I realised this had been copied from another file, not added new. A bunch of these comments are probably out of scope if things were just moved around.